crates/surrealdb-types/src/value/duration.rs

```diff
@@ -0,0 +1,263 @@
+		}
+		// Calculate the total years
+		let year = secs / SECONDS_PER_YEAR;
+		let secs = secs % SECONDS_PER_YEAR;
+		// Calculate the total weeks
+		let week = secs / SECONDS_PER_WEEK;
+		let secs = secs % SECONDS_PER_WEEK;
+		// Calculate the total days
+		let days = secs / SECONDS_PER_DAY;
+		let secs = secs % SECONDS_PER_DAY;
+		// Calculate the total hours
+		let hour = secs / SECONDS_PER_HOUR;
+		let secs = secs % SECONDS_PER_HOUR;
+		// Calculate the total minutes
+		let mins = secs / SECONDS_PER_MINUTE;
+		let secs = secs % SECONDS_PER_MINUTE;
+		// Calculate the total milliseconds
+		let msec = nano / NANOSECONDS_PER_MILLISECOND;
+		let nano = nano % NANOSECONDS_PER_MILLISECOND;
+		// Calculate the total microseconds
+		let usec = nano / NANOSECONDS_PER_MICROSECOND;
+		let nano = nano % NANOSECONDS_PER_MICROSECOND;
+		// Write the different parts
+		if year > 0 {
+			write!(f, "{year}y")?;
+		}
+		if week > 0 {
+			write!(f, "{week}w")?;
+		}
+		if days > 0 {
+			write!(f, "{days}d")?;
+		}
+		if hour > 0 {
+			write!(f, "{hour}h")?;
+		}
+		if mins > 0 {
+			write!(f, "{mins}m")?;
+		}
+		if secs > 0 {
+			write!(f, "{secs}s")?;
+		}
+		if msec > 0 {
+			write!(f, "{msec}ms")?;
+		}
+		if usec > 0 {
+			write!(f, "{usec}µs")?;
+		}
+		if nano > 0 {
+			write!(f, "{nano}ns")?;
+		}
+		Ok(())
+	}
+}
+
+impl std::ops::Add for Duration {
+	type Output = Self;
+	fn add(self, other: Self) -> Self {
+		match self.0.checked_add(other.0) {
+			Some(v) => Duration::from(v),
+			None => Duration::from(std::time::Duration::MAX),
+		}
+	}
+}
+
+impl<'b> std::ops::Add<&'b Duration> for &Duration {
+	type Output = Duration;
+	fn add(self, other: &'b Duration) -> Duration {
+		match self.0.checked_add(other.0) {
+			Some(v) => Duration::from(v),
+			None => Duration::from(std::time::Duration::MAX),
+		}
+	}
+}
+
+impl std::ops::Sub for Duration {
+	type Output = Self;
+	fn sub(self, other: Self) -> Self {
+		match self.0.checked_sub(other.0) {
+			Some(v) => Duration::from(v),
+			None => Duration::default(),
+		}
+	}
+}
+
+impl<'b> std::ops::Sub<&'b Duration> for &Duration {
+	type Output = Duration;
+	fn sub(self, other: &'b Duration) -> Duration {
+		match self.0.checked_sub(other.0) {
+			Some(v) => Duration::from(v),
+			None => Duration::default(),
+		}
+	}
+}
+
+impl std::ops::Add<Datetime> for Duration {
+	type Output = Datetime;
+	fn add(self, other: Datetime) -> Datetime {
+		match chrono::Duration::from_std(self.0) {
+			Ok(d) => match other.0.checked_add_signed(d) {
+				Some(v) => Datetime::from(v),
+				None => Datetime::default(),
+			},
+			Err(_) => Datetime::default(),
+		}
+	}
+}
+
+impl std::ops::Sub<Datetime> for Duration {
+	type Output = Datetime;
+	fn sub(self, other: Datetime) -> Datetime {
+		match chrono::Duration::from_std(self.0) {
+			Ok(d) => match other.0.checked_sub_signed(d) {
+				Some(v) => Datetime::from(v),
+				None => Datetime::default(),
+			},
+			Err(_) => Datetime::default(),
+		}
+	}
+}
```

Comment:
These traits have weird behavior when overflowing: if subtracting the duration from the datetime causes an overflow you suddenly get the current time. They should either panic on overflow or better be removed.